packages/twenty-server/src/database/commands/command-runners/__tests__/upgrade.command-runner.spec.ts

```diff
@@ -0,0 +1,381 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+
+import { SemVer } from 'semver';
+import { EachTestingContext } from 'twenty-shared';
+import { Repository } from 'typeorm';
+
+import { UpgradeCommandRunner } from 'src/database/commands/command-runners/upgrade.command-runner';
+import { EnvironmentVariables } from 'src/engine/core-modules/environment/environment-variables';
+import { EnvironmentService } from 'src/engine/core-modules/environment/environment.service';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { SyncWorkspaceMetadataCommand } from 'src/engine/workspace-manager/workspace-sync-metadata/commands/sync-workspace-metadata.command';
+
+class TestUpgradeCommandRunnerV1 extends UpgradeCommandRunner {
+  fromVersion = new SemVer('1.0.0');
+
+  public override async beforeSyncMetadataUpgradeCommandsToRun(): Promise<void> {
+    return;
+  }
+
+  public override async afterSyncMetadataUpgradeCommandsToRun(): Promise<void> {
+    return;
+  }
+}
+
+class InvalidVersionUpgradeCommandRunner extends UpgradeCommandRunner {
+  fromVersion = new SemVer('invalid');
+
+  protected async beforeSyncMetadataUpgradeCommandsToRun(): Promise<void> {
+    return;
+  }
+
+  protected async afterSyncMetadataUpgradeCommandsToRun(): Promise<void> {
+    return;
+  }
+}
+
+class TestUpgradeCommandRunnerV2 extends UpgradeCommandRunner {
+  fromVersion = new SemVer('2.0.0');
+
+  protected async beforeSyncMetadataUpgradeCommandsToRun(): Promise<void> {
+    return;
+  }
+
+  protected async afterSyncMetadataUpgradeCommandsToRun(): Promise<void> {
+    return;
+  }
+}
+
+type CommandRunnerValues =
+  | typeof TestUpgradeCommandRunnerV1
+  | typeof TestUpgradeCommandRunnerV2
+  | typeof InvalidVersionUpgradeCommandRunner;
+
+const generateMockWorkspace = (overrides?: Partial<Workspace>) =>
```

Comment:
can we re-use an existing mocked workspace and derived this one from it (just change the version)

Reply:
I could not find existing one, we could make this a reusable utils in the future let me know !